jupyterhub/tests/conftest.py

```diff
@@ -60,9 +59,6 @@ def pytest_collection_modifyitems(items):
     for item in items:
         if inspect.iscoroutinefunction(item.obj):
             item.add_marker('asyncio')
-        if hasattr(inspect, 'isasyncgenfunction'):
-            # double-check that we aren't mixing yield and async def
-            assert not inspect.isasyncgenfunction(item.obj)
```

Comment:
I think this assert wants to stay, no? Just the condition is always True?

Reply:
I added it back but removed the `hasattr(inspect, 'isasyncgenfunction')` part as it is added in py36 and we no longer support py36.

I'd like to enhance the comment to make me understand it better. I now better understand this as a test to ensure our test functions run by pytest, all prefixed with `test_`, not are async generator functions. I guess it may not make sense for them to be that, so this assertion is unrelated to the support for py36 support for async generator functions in general, right?

Reply:
Correct! The condition is always True for py36 so to leave the behavior unchanges would be to *always* execute the block without a condition instead of never executing it (which would be the py35 behavior).

The assertion was added specifically to catch folks used to writing tornado coroutines which use `yield` switching to `async def` functions. In particular, during the bulk migration in #2358. The reason I added it was that I saw that several tests I had not updated were *passing* without being run.

For instance, this test:

```python
async def test_fail():
    yield gen.sleep(1)
    assert False
```

would *pass*.

**However**, when I run with the latest versions of pytest, pytest-asyncio, this pattern does now fail:

```
E           TypeError: An asyncio.Future, a coroutine or an awaitable is required
```

so we can safely remove the check, as recent-enough pytest-asyncio handles this check itself.